src/query/service/src/interpreters/interpreter_table_drop.rs

```diff
@@ -73,6 +72,11 @@ impl Interpreter for DropTableInterpreter {
                 let latest = tbl.as_ref().refresh(self.ctx.as_ref()).await?;
                 latest.truncate(self.ctx.clone(), purge).await?
             }
+        } else {
+            return Err(ErrorCode::TableInfoError(format!(
```

Comment:
Hi @BohuTANG, I added this check because I think this branch in if condition is error prone. We don't know what will happen if Option<tbl> is empty, we just return  `Ok(PipelineBuildResult::create())`, which IMHO can be hard to debug in some edge cases. After I add this, a series of tests are failing, like https://github.com/datafuselabs/databend/actions/runs/4181126745/jobs/7242889308, can you pull in some one to explain a bit on, in which case tbl is empty? So that I can add some comment here to help explain the scenario.